app/controllers/posts_controller.rb

```diff
@@ -23,6 +23,17 @@ def create
     end
   end
 
+  def destroy
+    @post = Post.find(params[:id])
+    @comments = @post.comments
+    @likes = @post.likes
+    @comments.each(&:destroy)
+    @likes.each(&:destroy)
+    @post.destroy
+
+    redirect_to user_posts_url(current_user.id), notice: 'Post was successfully destroyed!'
+  end
+
```

Comment:
- [ ] Great job adding adding authorization rule to your project. It is required that `load_and_authorize_resource` method provided by cancancan gem above your post controller.
By using load_and_authorize_resource, CanCanCan will load the appropriate resource based on the controller and action name, and it will also authorize the action based on the defined abilities in your ability.rb file. If the user doesn't have the necessary permissions, it will raise a CanCan::AccessDenied exception that you can handle appropriately (e.g., by redirecting to an error page).
It should look something similar to this :point_down: 

```
    class PostsController < ApplicationController
       load_and_authorize_resource
       -
       -
       -
    end
```